xds/src/main/java/io/grpc/xds/XdsRouteConfigureResource.java

```diff
@@ -270,22 +266,19 @@ static StructOrError<Route> parseRoute(
               + routeMatch.getErrorDetail());
     }
 
-    Map<String, FilterConfig> overrideConfigs = Collections.emptyMap();
-    if (parseHttpFilter) {
-      StructOrError<Map<String, FilterConfig>> overrideConfigsOrError =
-          parseOverrideFilterConfigs(proto.getTypedPerFilterConfigMap(), filterRegistry);
-      if (overrideConfigsOrError.getErrorDetail() != null) {
-        return StructOrError.fromError(
-            "Route [" + proto.getName() + "] contains invalid HttpFilter config: "
-                + overrideConfigsOrError.getErrorDetail());
-      }
-      overrideConfigs = overrideConfigsOrError.getStruct();
+    StructOrError<Map<String, FilterConfig>> overrideConfigsOrError =
+        parseOverrideFilterConfigs(proto.getTypedPerFilterConfigMap(), filterRegistry);
+    if (overrideConfigsOrError.getErrorDetail() != null) {
+      return StructOrError.fromError(
+          "Route [" + proto.getName() + "] contains invalid HttpFilter config: "
+              + overrideConfigsOrError.getErrorDetail());
     }
```

Comment:
Same concern as with `XdsResourceType`, can this new check inadvertently break some users that don't use http filters?